src/main/java/school/faang/wow/Main.java

```diff
@@ -0,0 +1,30 @@
+package school.faang.wow;
+
+import java.util.concurrent.CompletableFuture;
+
+public class Main {
+    public static void main(String[] args) {
+        QuestSystem questSystem = new QuestSystem();
+
+        Player player1 = new Player("Thrall", 10, 200);
+        Player player2 = new Player("Sylvanas", 12, 450);
+
+        Quest quest1 = new Quest("Defeat the Lich King", DifficultyQuest.HARD, 100);
+        Quest quest2 = new Quest("Retrieve the Sword of Azeroth", DifficultyQuest.LEGEND, 100);
+
+        CompletableFuture<Player> player1Quest = null;
```

Comment:
изучи следующий код:

```suggestion
        List<CompletableFuture<Player>> quests = List.of(
                questSystem.startQuest(firstPlayer, firstQuest),
                questSystem.startQuest(firstPlayer, secondQuest),
                questSystem.startQuest(secondPlayer, secondQuest)
        );
        quests .stream()
                .peek(future -> future.thenAccept(questSystem::onQuestCompletion)
                        .exceptionally(er -> {
                            log.error(er.getMessage());
                            return null;
                        }))
                .map(CompletableFuture::join)
                .collect(Collectors.toSet());
```

Советую отказаться от хардкодинга, все делать через динамику, чтобы код работал как для 1 квеста, так и для N, в твоем случае у тебя идет перебор и переприсваивание переменной, что не очень. Кроме того, отсутствует обработка ошибок